modules/gdscript/gdscript_parser.cpp

```diff
@@ -4416,6 +4417,12 @@ bool GDScriptParser::export_annotations(const AnnotationNode *p_annotation, Node
 			push_error(_get_annotation_error_string(p_annotation->name, expected_types, variable->get_datatype()), p_annotation);
 			return false;
 		}
+	} else if (p_annotation->name == SNAME("@export_input")) {
+		if (!(variable->export_info.type == Variant::STRING_NAME || variable->export_info.type == Variant::STRING)) {
```

Comment:
I haven't tested, but it should be something like:

```suggestion
		if (!export_type.is_variant() && !(export_type.builtin_type == Variant::STRING_NAME || export_type.builtin_type == Variant::STRING)) {
```

Reply:
```suggestion
		if (!export_type.is_variant() && export_type.builtin_type != Variant::STRING_NAME && export_type.builtin_type != Variant::STRING) {
```
If the above is valid I'd say for clarity